### PR TITLE
Sphincs+

### PR DESCRIPTION
**Pull request by Soundes-M**

### CODE REVIEW

File: signatures/SPHINCS+/SPHINCS-Haraka/impl/avx2.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+name: AVX2
+platform: c
+type: optimized
+side channel guarding:
+  branching: true
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-Haraka/impl/avx2.yaml:5]: could add `timing: false for AES software implementation`

File: signatures/SPHINCS+/SPHINCS-SHA-256/impl/avx2.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+name: AVX2
+platform: c
+type: optimized
+side channel guarding:
+  branching: true
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/impl/avx2.yaml:5]: also `timing: true`

File: signatures/SPHINCS+/SPHINCS-SHA-256/param/128s-robust.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+name: 128s-robust
+security level:
+  classical:  128
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/param/128s-robust.yaml:3]: 133 classical security (NIST submission paper table 3)

File: signatures/SPHINCS+/SPHINCS-Haraka/param/128s-robust.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+name: 128s-robust
+security level:
+  classical:  128
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-Haraka/param/128s-robust.yaml:3]: 133 classical security (NIST submission paper table 3)

File: signatures/SPHINCS+/SPHINCS-Haraka/param/128s-simple.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+name: 128s-simple
+security level:
+  classical:  128
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-Haraka/param/128s-simple.yaml:3]: 133 classical security (NIST submission paper table 3)

File: signatures/SPHINCS+/SPHINCS-SHA-256/param/128s-simple.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+name: 128s-simple
+security level:
+  classical:  128
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/param/128s-simple.yaml:3]: 133 classical security (NIST submission paper table 3)

File: signatures/SPHINCS+/SPHINCS-SHAKE256/param/128s-robust.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+name: 128s-robust
+security level:
+  classical:  128
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/param/128s-robust.yaml:3]: 133 classical security (NIST submission paper table 3)

File: signatures/SPHINCS+/SPHINCS-SHAKE256/param/128s-simple.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+name: 128s-simple
+security level:
+  classical:  128
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/param/128s-simple.yaml:3]: 133 classical security (NIST submission paper table 3)

File: signatures/SPHINCS+/SPHINCS-SHAKE256/param/192f-robust.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+name: 192f-robust
+security level:
+  classical:  194
+  quantum:  97
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/param/192f-robust.yaml:5]: NIST level 3

File: signatures/SPHINCS+/SPHINCS-SHAKE256/param/192f-simple.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+name: 192f-simple
+security level:
+  classical:  194
+  quantum:  97
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/param/192f-simple.yaml:5]: NIST level 3

File: signatures/SPHINCS+/SPHINCS-SHAKE256/param/192s-robust.yaml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+name: 192s-robust
+security level:
+  classical: 196
+  quantum:  98
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/param/192s-robust.yaml:5]: NIST level 3

File: signatures/SPHINCS+/SPHINCS-SHAKE256/param/192s-simple.yaml [hunk 12]
```diff
@@ -0,0 +1,12 @@
+name: 192s-simple
+security level:
+  classical: 196
+  quantum:  98
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/param/192s-simple.yaml:5]: NIST level 3

File: signatures/SPHINCS+/SPHINCS-SHAKE256/param/256f-robust.yaml [hunk 13]
```diff
@@ -0,0 +1,12 @@
+name: 256f-robust
+security level:
+  classical:  254
+  quantum:  127
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/param/256f-robust.yaml:5]: NIST level 5

File: signatures/SPHINCS+/SPHINCS-SHAKE256/param/256f-simple.yaml [hunk 14]
```diff
@@ -0,0 +1,12 @@
+name: 256f-simple
+security level:
+  classical:  254
+  quantum:  127
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/param/256f-simple.yaml:5]: NIST level 5

File: signatures/SPHINCS+/SPHINCS-SHAKE256/param/256s-robust.yaml [hunk 15]
```diff
@@ -0,0 +1,12 @@
+name: 256s-robust
+security level:
+  classical:  255
+  quantum:  128
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/param/256s-robust.yaml:5]: NIST level 5

File: signatures/SPHINCS+/SPHINCS-SHAKE256/param/256s-simple.yaml [hunk 16]
```diff
@@ -0,0 +1,12 @@
+name: 256s-simple
+security level:
+  classical:  255
+  quantum:  128
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/param/256s-simple.yaml:5]: NIST level 5

File: signatures/SPHINCS+/SPHINCS-SHAKE256/impl/avx2.yaml [hunk 17]
```diff
@@ -0,0 +1,7 @@
+name: AVX2
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/impl/avx2.yaml:1]: Add link to paper here.

File: signatures/SPHINCS+/SPHINCS-SHAKE256/impl/avx2.yaml [hunk 18]
```diff
@@ -0,0 +1,7 @@
...
+platform: c
+type: optimized
+side channel guarding:
+  branching: true
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHAKE256/impl/avx2.yaml:5]: add `timing: true`

File: signatures/SPHINCS+/SPHINCS-SHA-256/param/192f-robust.yaml [hunk 19]
```diff
@@ -0,0 +1,12 @@
+name: 192f-robust
+security level:
+  classical:  194
+  quantum:  97
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/param/192f-robust.yaml:5]: NIST level 3

File: signatures/SPHINCS+/SPHINCS-SHA-256/param/192f-simple.yaml [hunk 20]
```diff
@@ -0,0 +1,12 @@
+name: 192f-simple
+security level:
+  classical:  194
+  quantum:  97
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/param/192f-simple.yaml:5]: NIST level 3

File: signatures/SPHINCS+/SPHINCS-SHA-256/param/192s-robust.yaml [hunk 21]
```diff
@@ -0,0 +1,12 @@
+name: 192s-robust
+security level:
+  classical: 196
+  quantum:  98
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/param/192s-robust.yaml:5]: NIST level 3

File: signatures/SPHINCS+/SPHINCS-SHA-256/param/192s-simple.yaml [hunk 22]
```diff
@@ -0,0 +1,12 @@
+name: 192s-simple
+security level:
+  classical: 196
+  quantum:  98
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/param/192s-simple.yaml:5]: NIST level 3

File: signatures/SPHINCS+/SPHINCS-SHA-256/param/256f-robust.yaml [hunk 23]
```diff
@@ -0,0 +1,12 @@
+name: 256f-robust
+security level:
+  classical:  254
+  quantum:  127
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/param/256f-robust.yaml:5]: NIST level 5

File: signatures/SPHINCS+/SPHINCS-SHA-256/param/256f-simple.yaml [hunk 24]
```diff
@@ -0,0 +1,12 @@
+name: 256f-simple
+security level:
+  classical:  254
+  quantum:  127
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/param/256f-simple.yaml:5]: NIST level 5

File: signatures/SPHINCS+/SPHINCS-SHA-256/param/256s-robust.yaml [hunk 25]
```diff
@@ -0,0 +1,12 @@
+name: 256s-robust
+security level:
+  classical:  255
+  quantum:  128
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/param/256s-robust.yaml:5]: NIST level 5

File: signatures/SPHINCS+/SPHINCS-SHA-256/param/256s-simple.yaml [hunk 26]
```diff
@@ -0,0 +1,12 @@
+name: 256s-simple
+security level:
+  classical:  255
+  quantum:  128
+  nist category: 1
```
Comment (taudor) [inline signatures/SPHINCS+/SPHINCS-SHA-256/param/256s-simple.yaml:5]: NIST level 5

**Comment by Soundes-M** [comment]
Updated

**Comment by taudor** [comment]
I fixed some small stuff, if your'e ok with it we can merge.

**Comment by Soundes-M** [comment]
> I fixed some small stuff, if your'e ok with it we can merge.

yep thanks!

**Pull request merged**